src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.Unix.cs

```diff
@@ -60,12 +64,31 @@ public sealed partial class TarWriter : IDisposable
 
             entry._header._mode = (status.Mode & 4095); // First 12 bits
 
-            entry.Uid = (int)status.Uid;
-            entry.Gid = (int)status.Gid;
+            // Uid and UName
+            entry._header._uid = (int)status.Uid;
+            _userIdentifiers ??= new Dictionary<uint, string>();
+            if (!_userIdentifiers.ContainsKey(status.Uid))
+            {
+                entry._header._uName = Interop.Sys.GetUName(status.Uid) ?? string.Empty;
+                _userIdentifiers.Add(status.Uid, entry._header._uName);
+            }
+            else
+            {
+                entry._header._uName = _userIdentifiers[status.Uid];
+            }
 
-            // TODO: Add these p/invokes https://github.com/dotnet/runtime/issues/68230
-            entry._header._uName = "";// Interop.Sys.GetUName();
-            entry._header._gName = "";// Interop.Sys.GetGName();
+            // Gid and GName
+            entry._header._gid = (int)status.Gid;
+            _groupIdentifiers ??= new Dictionary<uint, string>();
```

Comment:
This is called from WriteEntry.  Is there a meaningful scenario where a TarWriter is created but WriteEntry isn't called?  If that's not an expected case, this needn't be lazily initialized.